modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -545,19 +548,47 @@ class NDB_Form_Statistics extends NDB_Form
         } else {
             $Param_Project = '';
         }
-        $subprojects = Utility::getSubprojectsForProject($currentProject);
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
 
-        $this->tpl_data['Subprojects'] = $subprojects;
         foreach ($subprojects as $key=>$val) {
             $subprojList .= $key.",";
         }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubProject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if (isset($_REQUEST['BehaviouralSubproject'])
+            && $_REQUEST['BehaviouralSubproject'] != ''
+            && in_array($_REQUEST['BehaviouralSubproject'], array_keys($subprojects))
+        ) {
+            $currentSubProject = $_REQUEST['BehaviouralSubproject'];
+            $this->tpl_data['CurrentSubproject']
+                = array(
+                   'ID'   => $currentSubProject,
+                   'Name' => $subprojects[$currentSubProject],
+                  );
+            $suproject_query
+                = "AND (s.SubprojectID IS NULL".
+                " OR s.SubprojectID = ($currentSubProject))";
```

Comment:
@driusan `$currentSubProject` comes from `$_REQUEST['BehaviouralSubproject']` which comes from a drop down which only allows valid subprojects to be chosen
additionally i have
```
      if (isset($_REQUEST['BehaviouralSubproject'])
            && $_REQUEST['BehaviouralSubproject'] != ''
            && in_array($_REQUEST['BehaviouralSubproject'], array_keys($subprojects))
        ) {
```
which ensures the chosen option is valid
I basically follow the same method you wrote for choosing a valid Project
Unless I'm misunderstanding something, I don't see how the user would be able to set the value to something other than a valid subproject